### ISSUE TITLE
missing phpdoc after property initialization

### ISSUE THREAD

**Issue by fabioi** [kalessil/phpinspectionsea#250]
I think would be able to set a warning in case you initialize a property (eg: from a constructor) but not phpdoc has been set.

**Comment by rentalhost** [comment]
If I understand, I guess that it should be two issues:

1. Warn about no-phpdoc on class properties;
   1. Warn if `@type` is not set;
2. Warn if class uses (R/W) an undefined property (if it is magical, you should declares as `@property`);

**Comment by fabioi** [comment]
Hi thanks yes seems it. I am adding also another nice to have about improving the type check on a class property / field

**Comment by kalessil** [comment]
@rentalhost : #254, #252 are the same - property type declaration correctness. Would you please update/close tickets (so just 1 of them left)?

- Should check if property has PhpDoc
- Should check if property has type definition in PhpDoc;

**Comment by rentalhost** [comment]
My initial though about #252 was about describe what property was need (description).

```php
// Warning: you should describe that.
/** @type string */
protected $string;
```

**Comment by Fleshgrinder** [comment]
Are you sure that you want `@type` and not `@var`. The latter is much more widespread, although the former is more precise.

**Comment by funivan** [comment]
phpstorm has built-in inspection "Missing PHPDoc comment"

http://imagizer.imageshack.com/img923/4937/O6QwUX.png

**Comment by kalessil** [comment]
Perfect, closing this one then.
